prototype/local-factory/agents.go

```
@@ -0,0 +1,361 @@
+		args := []string{"exec", "-C", directory, "--ephemeral", "--color", "never", "--sandbox", "read-only"}
+		if role == "build" || role == "verify" {
+			args[len(args)-1] = "workspace-write"
+		}
+		if agent.Model != "" {
+			args = append(args, "--model", agent.Model)
+		}
+		args = append(args, "-")
+		command = exec.CommandContext(ctx, "codex", args...)
+	case "claude":
+		args := []string{"--print", "--safe-mode", "--no-session-persistence", "--output-format", "text", "--permission-mode", "dontAsk"}
+		if role == "foreman" {
```

golangci-lint (2.12.2)

[error] QF1003: could use tagged switch on role

(staticcheck)